pkg/operation/botanist/controlplane.go

```diff
@@ -252,9 +252,21 @@ func (b *Botanist) DeployClusterAutoscaler() error {
 	)
 
 	for _, worker := range b.MachineDeployments {
+		// Skip worker pools for which min=max=0.
+		if worker.Minimum == 0 && worker.Maximum == 0 {
+			continue
+		}
+
+		// Cluster Autoscaler requires min>=1. We ensure that in the API server validation part, however,
+		// for backwards compatibility, we treat existing worker pools whose minimum equals 0 as min=1.
```

Comment:
Hm, I guess we shouldn't, but we should mention it in the release notes and notify the users that they should change their `Shoot` resources to not have `min=0` in any of their worker pools if they want to use the cluster autoscaler reliably.
@mvladev any thoughts?

Reply:
The correct place to put this [functionality](https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#alpha-field-in-existing-api-version) in the Rest storage strategy for Shoot. Otherwise other clients of the gardener can be broken. After some time we can remove it. Something like:


```go
func (frobberStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
  frobber := obj.(*api.Frobber)

  if !utilfeature.DefaultFeatureGate.Enabled(features.Frobber2D) {
    frobber.Width = nil
  }
}

func (frobberStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, old runtime.Object) {
  newFrobber := obj.(*api.Frobber)
  oldFrobber := old.(*api.Frobber)

  if !utilfeature.DefaultFeatureGate.Enabled(features.Frobber2D) {
    newFrobber.Width = nil
    oldFrobber.Width = nil
  }
}
```

Reply:
Didn't get it, sorry. What exactly would you put in these two functions?

Reply:
I was mistaken - for this change is not needed. However, the problem with this change is:

- User has already created `Shoot` with `autoscaller` = true and `autoScalerMin` = 1
- API server is upgraded with this change
- Any changes to this resource - `PUT` / `PATCH` will fail.

Reply:
Yes, the users who want auto-scaling will have to set `min=1` for all their worker groups unless `min=max=0` (current trigger for cluster hibernation, planned to be changed to be triggered by another field in the future) if they want to change anything.
Since auto-scaling was not integrated before I guess that's no too hard restriction/problem. What do you think?

Reply:
LGMT